src/EPR.PRN.Backend.API.UnitTests/Validators/PrnControllerEndpointValidationTests.cs

```diff
@@ -0,0 +1,136 @@
+            matchingProp.Should().NotBeNull();
+
+            // Set the value of the property (overriding the default value set above) to the value passed in as the argument to this method
+            matchingProp.SetValue(dto, propertyValue);
+
+            // Act
+            var result = validator.Validate(dto);
+
+            // Assert
+            result.Should().NotBeNull();
+            result.Errors.Select(x => x.PropertyName).Should().Contain(propertyName);
+        }
+
+        [TestMethod]
+        [DataRow(true)]
+        [DataRow(false)]
+        public void Test_SavePrnDetailsRequestValidator_Returns_Error_WhenCancellationDateNotCorrectlySet(bool isCancelledStatus)
+        {
+            var validator = new SavePrnDetailsRequestValidator();
+
+            var dto = new SavePrnDetailsRequest()
+            {
+                AccreditationNo = "ABC",
+                AccreditationYear = 2018,
+                DecemberWaste = true,
+                EvidenceMaterial = "Aluminium",
+                EvidenceNo = Guid.NewGuid().ToString(),                
+                EvidenceTonnes = 5000,
+                ExternalId = Guid.NewGuid(),
+                IssueDate = DateTime.UtcNow.AddDays(-5),
+                IssuedByNPWDCode = "NPWD367742",
+                IssuedByOrgName = "ANB",
+                IssuedToEPRId = Guid.NewGuid(),
+                IssuedToNPWDCode = "NPWD557742",
+                IssuedToOrgName = "ZNZ",
+                IssuerNotes = "no notes",
+                IssuerRef = "ANB-1123",
+                MaterialOperationCode = "R-PLA",
+                ObligationYear = 2025,
+                PrnSignatory = "Pat Anderson",
+                PrnSignatoryPosition = "Director",
+                ProducerAgency = "TTL",
+                RecoveryProcessCode = "N11",
+                ReprocessorAgency = "BEX",
+                StatusDate = DateTime.UtcNow,
+            };
+
+            // Set Status based on test method parameter
+            if (isCancelledStatus)
+            {
+                dto.EvidenceStatusCode = Common.Enums.PrnStatus.Cancelled;
+                dto.CancelledDate = null;
+            }
+            else
+            {
+                dto.EvidenceStatusCode = Common.Enums.PrnStatus.AwaitingAcceptance;
+                dto.CancelledDate = DateTime.UtcNow.AddDays(-1);
+            }
+
+            // Act
+            var result = validator.Validate(dto);
+
+            // Assert
+            result.Should().NotBeNull();
+            
+            var errorPropertyNames = result.Errors.Select(x => x.PropertyName);
+            var cancelledDatePropertyName = nameof(dto.CancelledDate);
+
+            _ = isCancelledStatus 
+                ? errorPropertyNames.Should().Contain(cancelledDatePropertyName)
+                : errorPropertyNames.Should().NotContain(cancelledDatePropertyName);
+        }
+    }
+}
```

Comment:
Can we add one more test just to confirm we are not getting any validation errors if all attributes are set as expected?

Reply:
added